### PR TITLE
Use valid syntax for `T.attached_class` in suggseted sig

### LINKED ISSUE #7004: Bad autocorrect: `T.attached_class (of ...)` in sig
#### Input

[→ View on sorbet.run](https://sorbet.run/#%23%20typed%3A%20strict%0Aextend%20T%3A%3ASig%0A%0Aclass%20Foo%0A%20%20def%20self.generate%0A%20%20%20%20self.new%28%29%0A%20%20end%0Aend)

```ruby
# typed: strict
extend T::Sig

class Foo
  def self.generate
    self.new()
  end
end
```

#### Observed output

```
editor.rb:5: The method generate does not have a sig https://srb.help/7017
     5 |  def self.generate
          ^^^^^^^^^^^^^^^^^
  Autocorrect: Use -a to autocorrect
    editor.rb:5: Insert sig { returns(T.attached_class (of Foo)) }
     5 |  def self.generate
          ^
    editor.rb:5: Insert extend T::Sig
     5 |  def self.generate
        ^
Errors: 1
```

#### Expected behavior

- - -

### PR DESCRIPTION
**Pull request by jez**
### Motivation

Fixes #7004

### Test plan

See included automated tests.

### CODE REVIEW

File: core/Symbols.cc [hunk 1]
```diff
@@ -1847,7 +1848,6 @@ void ClassOrModule::recordSealedSubclass(MutableContext ctx, ClassOrModuleRef su
     const OrType *orT = nullptr;
     while ((orT = cast_type<OrType>(*iter))) {
         auto right = cast_type_nonnull<ClassType>(orT->right);
-        ENFORCE(left);
```
Comment (neilparikh) [inline core/Symbols.cc:1850]: Why is this change here?

**Review by neilparikh — APPROVED** [review]

File: core/Symbols.cc (hunk 1 above)
Comment (jez) [reply core/Symbols.cc:1850]: I thought I committed this to a different branch, but it looks like when I was rebasing it got added to the wrong branch. I’ll fix tomorrow

Comment (jez) [reply core/Symbols.cc:1850]: interestingly, this didn't end up committed on two branches, and since it's a no-op I'm just going to land it now so I don't have to wait for a passing rebuild.

Comment (jez) [reply core/Symbols.cc:1850]: interestingly, this didn't end up committed on two branches, and since it's a no-op I'm just going to land it now so I don't have to wait for a passing rebuild.
